### DISCUSSION TITLE
recordId (id) shows as undefined when fetching an image

### DISCUSSION THREAD

**Discussion by tekojtrapkomo** [pocketbase/pocketbase#1792] [Q&A] [answered]
id shows as undefined in sveltekit, tried almost every params and no luck? what am i missing guys?

`const getGallery = async (recordId) => {
const gallery = NonPojos(await locals.pb.collection('gallery').getOne(recordId);
return gallery;
};`

**Comment by ganigeorgiev** [comment, marked as answer]
Are there any JS errors in the browser/node console (_note that in the above example there seems to be missing a closing bracket_)?

You can try wrapping the await with `try/catch` and console.log the error.

Also make sure that your client satisfies the gallery "View API" rule.

**Comment by tekojtrapkomo** [comment]
Hello @ganigeorgiev. 

Here is the full code snippet: 
**+page.server.js**
```
export const load = ({ locals }) => {
const getGallery = async (recordId) => {
try {
const gallery = NonPojos(await locals.pb.collection('gallery').getOne(recordId));
return gallery;
} catch (err) {
console.log('Error: ', err);
throw error(err.status, err.message);
}
};
```

**Svelte Component**
_Script Tag_
```
export let data;
export const imgURL = (collectionId,recordId, submission) => {
    return 'https://mypbserver/api/files/${collectionId}/${recordId}/${submission}';
}
```
_DOM_
```
{#each data.gallery as gallery}
        <div>
            <img src={imgURL(gallery.collectionId, gallery.recordId, gallery.submission)} alt="{gallery.title} by {gallery.user}">
            <p>{gallery.title}</p>
        </div>
{/each}
```
Collection ID and Image URL(submission) are working fine it's only the recordId. And console log of data returns submission as [array]

**Reply by ganigeorgiev** [reply]
I'm sorry. I don't understand the example. There is no such field `gallery.recordId` (unless `recordId` is a custom field in your collection). If you want to access the id of the gallery record you can just use `gallery.id`. You can `console.log(gallery)` to explore all available fields.

**Reply by tekojtrapkomo** [reply]
Working perfect! thank you so much
